Automattic/jetpack
# Add LiveChat icon to WPAdmin on Atomic only.

Adds the "FAB" LiveChat icon to WPAdmin on WordPress.com envs only. This is a non-interactive element as it is not yet possible to "embed" the actual FAB widget from Calypso into WPAdmin. This acts as a temporary workaround to allow the user access to "help" via a direct link to `wordpress.com/help`.

<img width="1278" alt="Screen Shot 2021-02-22 at 11 37 17" src="https://user-images.githubusercontent.com/444434/108703306-59170f80-7502-11eb-9fe3-df283a55b994.png">


To do this we must:

* Include the same markup as per Calypso FAB.
* Inline the Grid Icon SVG so as not to rely on Gridicons.
* Apply layout styles based on Calypso styles.

In addition we must use the existing color schemes unification process to define custom color overides for the icon depending on which color scheme is being used in WPAdmin. This involves adding new vars to each color scheme file and then applying these to the FAB styling.

Fixes https://github.com/Automattic/wp-calypso/issues/50271
Fixes https://github.com/Automattic/wp-calypso/issues/50367

#### Please note

* I discussed this with @jeherve and we decided putting in Masterbar was an ok choice - at least for the sake of expediency for now.
* We are going to ignore custom color schemes for the LiveChat icon for now. We can follow up to add color scheme support later as required.

## Changes proposed in this Pull Request:
* Add LiveChat icon to WPAdmin for Atomic only.

## Jetpack product discussion

See paYJgx-1nM-p2

#### Does this pull request change what data or activity we track or use?
No

## Testing instructions:

The following steps should be taken on both Simple & Atomic (the visual icon check should be tested across a variety of screen sizes):

* Go to WPAdmin. 
* See LiveChat icon in bottom right.
* Click icon to be taken to `wordpress.com/help`.
* Go to `wordpress.com` and open your site in _Calypso_.
* Create a new Post and check you are within the Gutenframed editor (here's how you check for that PCYsg-lwV-p2).
* See only a _single_ icon in bottom right - it should be the proper Calypso FAB that you can interact with and not the static version.
* Go back to WPAdmin and create a new post **_not_** using the Gutenframed editor. 
* Again you should see a single icon in the bottom right but it should be the static version that is just a link and not the interactive Calypso version.

#### Color Schemes
* Also try changing your color scheme and then re-testing to ensure the icon picks up the color schemes correctly.


### Atomic sites

* Get Atomic site with JP Beta.
* Checkout this branch using JP Beta.
* Follow testing steps above.

### Simple sites

This doesn't work yet because we need to update JP Fusion to sync the `inline-help` directory within masterbar to Dotcom Simple codebase.

Once this PR is provisionally accepted for Atomic sites then we'll merge this diff (D57403-code) to update JP sync and then rebase all the things.

* Follow testing steps above.



## Proposed changelog entry for your changes:


* Adds WP.com Live Chat icon (link only) to bottom right of WPAdmin for Atomic sites only.
